src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/EverythingOperationHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Core.Features.Security.Authorization;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Features.Security;
+using Microsoft.Health.Fhir.Core.Messages.Everything;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Everything
+{
+    public class EverythingOperationHandler : IRequestHandler<EverythingOperationRequest, EverythingOperationResponse>
+    {
+        private readonly IPatientEverythingService _patientEverythingService;
+        private readonly IBundleFactory _bundleFactory;
+        private readonly IAuthorizationService<DataActions> _authorizationService;
+
+        public EverythingOperationHandler(IPatientEverythingService patientEverythingService, IBundleFactory bundleFactory, IAuthorizationService<DataActions> authorizationService)
+        {
+            EnsureArg.IsNotNull(patientEverythingService, nameof(patientEverythingService));
+            EnsureArg.IsNotNull(bundleFactory, nameof(bundleFactory));
+            EnsureArg.IsNotNull(authorizationService, nameof(authorizationService));
+
+            _patientEverythingService = patientEverythingService;
+            _bundleFactory = bundleFactory;
+            _authorizationService = authorizationService;
+        }
+
+        public async Task<EverythingOperationResponse> Handle(EverythingOperationRequest request, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            if (await _authorizationService.CheckAccess(DataActions.Read, cancellationToken) != DataActions.Read)
+            {
+                throw new UnauthorizedFhirActionException();
+            }
+
+            SearchResult searchResult = await _patientEverythingService.SearchAsync(
```

Comment:
>searchResult [](http://example.com/codeflow?start=25&length=12)

Reading code I have feeling what EverythingOperationHandler would eventually support all 4 operations. I would add if check (or switch) to check request.ResourceType and call `_patientEverythingService` if it's `Patient` and throw exception with (`resourceType` not supported for $everything operation or something like that).

Reply:
Yes, I have the same feeling that `EverythingOperationHandler` will eventually support all the required $everything operation. A check is added here. If `EverythingOperationType` is not Patient, `RequestNotValidException` is thrown.